api/v1beta1/verticareplicator_types.go

```diff
@@ -43,6 +43,45 @@ type VerticaReplicatorSpec struct {
 	// be enabled on the source database. Custom username for source and target
 	// databases is not supported yet when TLS configuration is used.
 	TLSConfig string `json:"tlsConfig,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=async
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Information of the target Vertica database to replicate to
+	Mode string `json:"mode"`
+}
+
+type VerticaReplicatorSourceDatabaseInfo struct {
+	VerticaReplicatorDatabaseInfo `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// The object name we want to copy from the source side. The available types are: namespace, schema, table.
+	// If this is omitted, the operator will replicate all namespaces in the source database.
+	ObjectName string `json:"objectName"`
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// A string containing a wildcard pattern of the schemas and/or tables to include in the replication.
+	// Namespace names must be front-qualified with a period.
+	IncludePattern string `json:"includePattern"`
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// A string containing a wildcard pattern of the schemas and/or tables to exclude from the set of tables matched
+	// by the include pattern. Namespace names must be front-qualified with a period.
+	ExcludePattern string `json:"excludePattern"`
```

Comment:
```suggestion
	ObjectName string `json:"objectName,omitempty"`
	// +kubebuilder:validation:Optional
	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
	// A string containing a wildcard pattern of the schemas and/or tables to include in the replication.
	// Namespace names must be front-qualified with a period.
	IncludePattern string `json:"includePattern,omitempty"`
	// +kubebuilder:validation:Optional
	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
	// A string containing a wildcard pattern of the schemas and/or tables to exclude from the set of tables matched
	// by the include pattern. Namespace names must be front-qualified with a period.
	ExcludePattern string `json:"excludePattern,omitempty"`
```
For backward compatibility.